network/gossip/libp2p/topology/topology_test.go

```diff
@@ -0,0 +1,173 @@
+	ids = ids.Union(collector)
+	ids = ids.Union(access)
+	ids = ids.Union(consensus)
+	ids = ids.Union(verification)
+	ids = ids.Union(execution)
+
+	// mocks state for collector nodes topology
+	state, _ := topology.CreateMockStateForCollectionNodes(suite.T(),
+		ids.Filter(filter.HasRole(flow.RoleCollection)), uint(cluster))
+
+	subMngrs := topology.MockSubscriptionManager(suite.T(), ids)
+
+	return state, ids, subMngrs
+}
+
+// multiSystemEndToEndConnectedness is a test helper evaluates end-to-end connectedness of the system graph
+// over several number of systems each with specified number of nodes on each role.
+func (suite *TopologyTestSuite) multiSystemEndToEndConnectedness(system, acc, col, con, exe, ver, cluster int) {
+	// creates a histogram to keep average fanout of nodes in systems
+	var aveHist *thist.Hist
+	if suite.printTrace() {
+		aveHist = thist.NewHist(nil, fmt.Sprintf("Average fanout for %d systems", system), "fit", 10, false)
+	}
+
+	for j := 0; j < system; j++ {
+		// adjacency map keeps graph component of a single channel ID
+		adjMap := make(map[flow.Identifier]flow.IdentityList)
+
+		// creates a flow system
+		state, ids, subMngrs := suite.generateSystem(acc, col, con, exe, ver, cluster)
+
+		var systemHist *thist.Hist
+		if suite.printTrace() {
+			// creates a fanout histogram for this system
+			systemHist = thist.NewHist(nil, fmt.Sprintf("System #%d fanout", j), "auto", -1, false)
+		}
+
+		totalFanout := 0 // keeps summation of nodes' fanout for statistical reason
+
+		// creates topology of the nodes
+		for i, id := range ids {
+			fanout := suite.topologyScenario(id.NodeID, subMngrs[i], ids, state)
+			adjMap[id.NodeID] = fanout
+
+			if suite.printTrace() {
+				systemHist.Update(float64(len(fanout)))
+			}
+			totalFanout += len(fanout)
+		}
+
+		if suite.printTrace() {
+			// prints fanout histogram of this system
+			fmt.Println(systemHist.Draw())
+			// keeps track of average fanout per node
+			aveHist.Update(float64(totalFanout) / float64(len(ids)))
+		}
+
+		// checks end-to-end connectedness of the topology
+		topology.CheckConnectedness(suite.T(), adjMap, ids)
+	}
+
+	if suite.printTrace() {
+		fmt.Println(aveHist.Draw())
+	}
+}
+
+// topologyScenario is a test helper that creates a StatefulTopologyManager with the LinearFanoutFunc,
+// it creates a TopicBasedTopology for the node and returns its fanout.
+func (suite *TopologyTestSuite) topologyScenario(me flow.Identifier,
+	subMngr channel.SubscriptionManager,
+	ids flow.IdentityList,
+	state protocol.ReadOnlyState) flow.IdentityList {
+
+	// creates topology of the node
+	top, err := topology.NewTopicBasedTopology(me, state, subMngr)
+	require.NoError(suite.T(), err)
+
+	// generates topology of node
+	myFanout, err := top.GenerateFanout(ids)
+	require.NoError(suite.T(), err)
+
+	return myFanout
+}
+
+// printTrace returns true if local environment variable Trace is found.
+func (suite *TopologyTestSuite) printTrace() bool {
```

Comment:
```suggestion
func (suite *TopologyTestSuite) shouldPrintTrace() bool {
```

Otherwise it sounds like this function is printing the trace